File: python/src/grains.py
SQUARE_COUNT = 64

def square(number):
    if number not in range(1, SQUARE_COUNT+1):
        raise ValueError(f"square must be between 1 and {SQUARE_COUNT}")
    return 2 ** (number - 1)


def total():
    return 2 ** SQUARE_COUNT - 1

File: typescript/src/grains.ts
export const square = (value: number) => {
  if (value < 1 || value > 64) throw Error("value should be between 1 and 64.")
  return 2n ** BigInt(value - 1)
}

export const total = () => {
  return 2n ** BigInt(64) - 1n
}
